File: recover.js
const fs = require('fs');

const logPath = 'C:\\Users\\Perfect Elect\\.gemini\\antigravity\\brain\\58db37e6-a496-4b95-80a8-c48b97514189\\.system_generated\\logs\\transcript.jsonl';
const outPath = 'c:\\Users\\Perfect Elect\\Desktop\\Website Projects\\Anmar logistics\\index.html';

const fileLines = fs.readFileSync(logPath, 'utf-8').split('\n');
const linesDict = {};

for (const line of fileLines) {
  if (!line.trim()) continue;
  try {
    const obj = JSON.parse(line);
    if (obj.type === 'TOOL_RESPONSE' && obj.status === 'DONE') {
      const content = obj.content || '';
      
      // The content of the tool response is what the wrapper gave to the model
      // Example: 
      // Created At: ...
      // File Path: ...
      // Total Lines: ...
      // Showing lines 1 to 800
      // The following code has been modified...
      // 1: <!DOCTYPE html>
      // 2: ...
      
      if (content.includes('Showing lines') && content.includes('The following code has been modified')) {
        const lines = content.split('\n');
        let parsing = false;
        
        for (const l of lines) {
          if (l.includes('The following code has been modified')) {
            parsing = true;
            continue;
          }
          if (l.includes('The above content does NOT show the entire file')) {
            parsing = false;
          }
          if (l.includes('The above content shows the entire')) {
            parsing = false;
          }
          
          if (parsing) {
            const match = l.match(/^(\d+): (.*)$/);
            if (match) {
              const num = parseInt(match[1]);
              const text = match[2];
              linesDict[num] = text;
            }
          }
        }
      }
    }
  } catch (e) {
    // ignore
  }
}

const keys = Object.keys(linesDict).map(Number);
if (keys.length > 0) {
  const max = Math.max(...keys);
  let out = '';
  for (let i = 1; i <= max; i++) {
    out += (linesDict[i] || '') + '\n';
  }
  fs.writeFileSync(outPath, out);
  console.log(`Successfully recovered ${keys.length} lines up to max ${max}`);
} else {
  console.log('Failed to find lines.');
}

File: recover.py
import json, re

log_path = r"C:\Users\Perfect Elect\.gemini\antigravity\brain\58db37e6-a496-4b95-80a8-c48b97514189\.system_generated\logs\transcript.jsonl"
out_path = r"c:\Users\Perfect Elect\Desktop\Website Projects\Anmar logistics\index.html"

lines_dict = {}

try:
    with open(log_path, "r", encoding="utf-8") as f:
        for line in f:
            try:
                data = json.loads(line)
                if data.get("type") == "TOOL_RESPONSE" and data.get("status") == "DONE":
                    content = data.get("content", "")
                    # We are looking for the view_file outputs of index.html
                    if "Total Lines: 1793" in content and "Showing lines" in content:
                        for text_line in content.split("\n"):
                            # Match the line number prefix exactly
                            match = re.match(r"^(\d+): (.*)", text_line)
                            if match:
                                line_num = int(match.group(1))
                                line_text = match.group(2)
                                lines_dict[line_num] = line_text
            except Exception:
                pass

    if len(lines_dict) >= 1790:
        with open(out_path, "w", encoding="utf-8") as out:
            for i in range(1, max(lines_dict.keys()) + 1):
                out.write(lines_dict.get(i, "") + "\n")
        print(f"Success! Recovered {len(lines_dict)} lines.")
    else:
        print(f"Failed. Found {len(lines_dict)} lines.")
        # let's print the keys we have
        print(sorted(list(lines_dict.keys()))[:10])
except Exception as e:
    print(f"Error: {e}")
